### PR TITLE
[shopsys] change forms in popup so they work with js validation

### LINKED ISSUE #361: Javascript validation of form in popup window
Reported by @swejzi and @jDolba.
### What is happening
When there are some javascript validation errors while submitting a form, a popup window with list of the errors is displayed. This is standard feature for all forms. But when the form is displayed in popup window, then the popup with errors overwrites the original window with the form. Normally, this is also wanted feature as we do not want to open multiple popups. 

### Expected result
Javascript validation of form in popup is resolved somehow nicely from UX point of view. The solution should be discussed with @LukasHeinz beforehand.

### PR DESCRIPTION
**Pull request by DavidOstrozlik**
| Q             | A
| ------------- | ---
|Description, reason for the PR| JS validation doesn't work with forms in popups, because it closes the original popup to show errors
|New feature| No 
|BC breaks| Yes 
|Fixes issues| closes #361  
|Standards and tests pass| No
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

When solving #361 I found out that there are 3 problems, that should be solved.

1. We don't have any form in popup with working JS validation
Solution: I changed login form in popup so it is loaded by AJAX and therefore works with JS validation

2. JS validation opens popup with error messages that closes popup with form
Solution: I Added hidden div to popup and changed `showFormErrorsWindow` function in a way that it show errors in the hidden div if popup is open

3. The errors in popup should be under the heading
Solution: I added heading option to window.js so you can have heading at the top of the popup, then you have error messages and then you have the content of the popup

### CODE REVIEW

File: project-base/src/Shopsys/ShopBundle/Resources/scripts/frontend/login.js [hunk 1]
```diff
@@ -3,9 +3,30 @@
...
+        };
+
+        function showWindow (event) {
+            Shopsys.ajax({
+                url: $(this).data('url'),
+                type: 'POST',
+                success: function (data) {
+                    var $window = Shopsys.window({
+                        content: data,
+                        textHeading: Shopsys.translator.trans('Login')
```
Comment (PetrHeinz) [inline project-base/src/Shopsys/ShopBundle/Resources/scripts/frontend/login.js]: The feature you're using here is not yet implemented (it belongs to the next commit).

File: project-base/src/Shopsys/ShopBundle/Form/Front/Login/LoginFormType.php [hunk 2]
```diff
@@ -51,8 +53,9 @@ public function getBlockPrefix()
      */
     public function configureOptions(OptionsResolver $resolver)
     {
-        $resolver->setDefaults([
-            'attr' => ['novalidate' => 'novalidate'],
-        ]);
+        $resolver
+            ->setDefaults([
+                'attr' => ['novalidate' => 'novalidate'],
+            ]);
```
Comment (PetrHeinz) [inline project-base/src/Shopsys/ShopBundle/Form/Front/Login/LoginFormType.php]: No change in this file was actually necessary.

File: project-base/src/Shopsys/ShopBundle/Resources/scripts/frontend/window.js [hunk 3]
```diff
@@ -165,6 +171,7 @@
         }
 
         Shopsys.register.registerNewContent($window);
+        $window.load();
```
Comment (PetrHeinz) [inline project-base/src/Shopsys/ShopBundle/Resources/scripts/frontend/window.js]: To be honest, I'm not sure why are you calling the method. I'm not even sure which method it actually is, I found 2 `.load()` methods in jQuery (https://api.jquery.com/load/ and deprecated https://api.jquery.com/load-event/), both require at least one argument.

File: docs/upgrade/UPGRADE-unreleased.md [hunk 4]
```diff
@@ -159,6 +159,27 @@ for instance:
     - if you have extended `CountryController` revise your changes – `new` and `edit` actions were added
 - if you have extended `Localization` class, you have to add type-hints to extended methods because they were added in the parent class ([#806](https://github.com/shopsys/shopsys/pull/806))
     - if you have extended method `Localization::getAdminLocale()` only to have administration in a different language than english, you can delete it and set parameter `shopsys.admin_locale` in your `parameters.yml` file instead
+- changed JS validation of forms in popup and login form in popup is now get by AJAX ([#782](https://github.com/shopsys/shopsys/pull/782))
```
Comment (TomasLudvik) [inline docs/upgrade/UPGRADE-unreleased.md]:
```suggestion
- fixed JS validation of forms in popup windows ([#782](https://github.com/shopsys/shopsys/pull/782))
    - login form in popup is now loaded via AJAX
```

File: docs/upgrade/UPGRADE-unreleased.md [hunk 5]
```diff
@@ -159,6 +159,27 @@ for instance:
...
+    - in `window.js` add options `textHeading = ''` and `cssClassHeading: ''` to `var defaults` and then add it to `$windowContent` and add div with `js-validation-errors` class to every popup.
```
Comment (TomasLudvik) [inline docs/upgrade/UPGRADE-unreleased.md]: I would change this to: 
This javascript files has been changed, please update them appropriately:
- src/Shopsys/ShopBundle/Resources/scripts/frontend/login.js
- src/Shopsys/ShopBundle/Resources/scripts/frontend/window.js

For me is better to see diff on Github than this instructions

File: docs/upgrade/UPGRADE-unreleased.md [hunk 6]
```diff
@@ -159,6 +159,27 @@ for instance:
...
+                      $windowContent.append('<h2 class="' + options.cssClassHeading + '">' + options.textHeading + '</h2>');
+                  }
+
+                  $windowContent.append(
+                      '<div class="display-none in-message in-message--alert js-window-validation-errors"></div>'
+                      + options.content
+                  );
+            ```
+- *(optional)* change login form so it is loaded by AJAX and works with JS validation
+    - you can change it as it was done in this [commit](https://github.com/shopsys/shopsys/pull/782/commits/bb6dadfce1d588aa3e06a8ba7b117d3de7afc1ce)
```
Comment (TomasLudvik) [inline docs/upgrade/UPGRADE-unreleased.md]: You did not mention changes in templates

File: packages/framework/src/Resources/scripts/common/validation/customizeBundle.js [hunk 7]
```diff
@@ -441,14 +442,26 @@
 
     Shopsys.validation.showFormErrorsWindow = function (container) {
         var $formattedFormErrors = Shopsys.validation.getFormattedFormErrors(container);
+        var $window = $('#js-window');
+
+        if ($window.length === 0) {
+            Shopsys.window({
+                content:
+                    '<div class="text-left">'
```
Comment (TomasLudvik) [inline packages/framework/src/Resources/scripts/common/validation/customizeBundle.js]: I would refactor this part of code, because it is identical as code few lines bellow and it violation with DRY principle

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Login/loginForm.html.twig [hunk 8]
```diff
@@ -15,9 +15,24 @@
 
         <h1>{{ 'Login'|trans }}</h1>
 
-        {{ form_row(form.email, {label: 'E-mail'|trans} ) }}
 
-        {{ form_row(form.password, {label: 'Password'|trans} ) }}
+        <dl class="form-line">
```
Comment (TomasLudvik) [inline project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Login/loginForm.html.twig]: Why are these changes necessary

Comment (DavidOstrozlik) [reply project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Login/loginForm.html.twig]: Because form attribute `multipleForm: true` changes ID's only for form labels and form rows but not for form_errors so we need remove the form errors so JS validation in login popup works on /login page

File: docs/upgrade/UPGRADE-unreleased.md (hunk 5 above)
Comment (DavidOstrozlik) [reply docs/upgrade/UPGRADE-unreleased.md]: The change or login.js is mentioned in the optional point below and It is only few lines so I think it's more readable here

Comment (TomasLudvik) [reply docs/upgrade/UPGRADE-unreleased.md]: We have discussed it and you want to make it more straightforward
